dcmdata/libsrc/dcistrmf.cc

```diff
@@ -257,3 +257,112 @@ DcmInputStreamFactory *DcmInputTempFileStreamFactory::clone() const
 {
     return new DcmInputTempFileStreamFactory(*this);
 }
+
+/* ============================================================================== */
+
+// DcmStdinProducer class is responsible for reading the binary data from a Dcm file
+// and is triggered when the filename is '-'
+DcmStdinProducer::DcmStdinProducer(const OFFilename &filename, offile_off_t offset)
+: DcmProducer()
+, file_()
+, status_(EC_Normal)
+, size_(0)
+{
+
+    char ch;
+     // Get the number of bytes in the file
+     while ( std::cin >> std::noskipws >> ch) {
+	     arr.push_back(ch);
+	 
+     }
+     
+     size_ = arr.size();
+
+     position = offset;
+}
+
+DcmStdinProducer::~DcmStdinProducer()
+{
+}
+
+OFBool DcmStdinProducer::good() const
+{
+  return status_.good();
+}
+
+OFCondition DcmStdinProducer::status() const
+{
+  return status_;
+}
+
+OFBool DcmStdinProducer::eos()
+{
+  if (position == arr.size() ) {
+      return OFTrue;
+  }
+  return OFFalse;
+}
+
+offile_off_t DcmStdinProducer::avail()
+{
+  return size_ - 1 - position;
```

Comment:
I'm reading it very fast obviously, but for binary stream of data, why do you need  - 1?